core/sql/executor/ExHbaseAccess.cpp

```diff
@@ -1645,12 +1650,18 @@ Lng32 ExHbaseAccessTcb::createSQRowFromAlignedFormat(Int64 *latestRowTimestamp)
   
   if (convertExpr())
     {
+      convertRowLen_ = hbaseAccessTdb().convertRowLen();
+      UInt32 rowLen = convertRowLen_;
       ex_expr::exp_return_type evalRetCode =
-	convertExpr()->eval(pentry_down->getAtp(), workAtp_, NULL, asciiRowLen);
+	convertExpr()->eval(pentry_down->getAtp(), workAtp_, NULL, 
+			    asciiRowLen, &rowLen);
       if (evalRetCode == ex_expr::EXPR_ERROR)
 	{
 	  return -1;
 	}
+      if (hbaseAccessTdb().getUseCif() &&
+      rowLen < convertRowLen_)
+        convertRowLen_=rowLen;
```

Comment:
Will the rowLen be less than convertRowLen from the compiler. I believe convertRowLen_ might have been ujsed to pre-allocate row buffer. There could be buffer overflow. Can you please confirm that is not the case.